src/Castle.Core.Tests/Castle.Core.Tests.csproj

```diff
@@ -3,7 +3,14 @@
 	<Import Project="..\..\buildscripts\common.props"></Import>
 
 	<PropertyGroup>
-		<TargetFrameworks>net462;netcoreapp2.1;netcoreapp3.1</TargetFrameworks>
+		<TargetFrameworks>net462;netcoreapp2.1;netcoreapp3.1;net6.0</TargetFrameworks>
+
+		<NoWarn>$(NoWarn);SYSLIB0003</NoWarn>
+		<!--
+			Code Access Security is not available in Core, and obsolete in net5.0+
+			This stops warnings about SYSLIB0003 (https://docs.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0003)
+			Depending on how these are deprecated in a future dotnet release, may need to wrap relevant code with "#if NET462_OR_GREATER" or similar at some point.
+		-->
```

Comment:
I agree with @stakx, surprised to see this.

Could this be changed to `#pragma warning disable` around the code rather than here so that we can see exactly what's using it and so more code isn't added.

I guess it'll at least be this unit test:
https://github.com/castleproject/Core/blob/d88e94447150996973d947a150ffda5efc0c3c1c/src/Castle.Core.Tests/DynamicProxy.Tests/AttributesToAvoidReplicatingTestCase.cs#L77-L80

Reply:
There's one instance in `Castle.Core`, and a few in `Castle.Core.Tests`:
https://github.com/castleproject/Core/blob/d88e94447150996973d947a150ffda5efc0c3c1c/src/Castle.Core/DynamicProxy/Generators/AttributesToAvoidReplicating.cs#L35

https://github.com/castleproject/Core/blob/d88e94447150996973d947a150ffda5efc0c3c1c/src/Castle.Core.Tests/DynamicProxy.Tests/AttributesToAvoidReplicatingTestCase.cs#L74

https://github.com/castleproject/Core/blob/d88e94447150996973d947a150ffda5efc0c3c1c/src/Castle.Core.Tests/DynamicProxy.Tests/AttributesToAvoidReplicatingTestCase.cs#L86

https://github.com/castleproject/Core/blob/d88e94447150996973d947a150ffda5efc0c3c1c/src/Castle.Core.Tests/DynamicProxy.Tests/AttributesToAvoidReplicatingTestCase.cs#L89

Happy to move the suppression to be `#pragma warning disable SYSLIB0001` if that's preferable - for the test file, would you want it for each instance, or just once at the top of the file?